infra/Pos/Diffusion/Subscription/Dns.hs

```diff
@@ -81,16 +88,17 @@ dnsSubscriptionWorker oq networkCfg DnsDomains{..} keepaliveTimer nextSlotDurati
             pure (dnsPeers', ())
         -- Try to subscribe to some peer.
         -- If they all fail, wait a while before trying again.
-        subscribeToOne dnsPeersList
-        retryInterval >>= delay
-        subscribeAlts dnsPeersVar (index, alts)
+        subscribeToOne dnsPeersList subDuration
+        d <- swapMVar subDuration 0
+        retryInterval d >>= delay
```

Comment:
It is technically out of the scope of this patch, but I think it'd be good to put a `logNotice` here which includes the actual delay. `dnsSubscriptionWorker: waiting <n>ms before trying again`.